internal/v4/auth.go

```diff
@@ -227,14 +263,22 @@ func (h *ReqHandler) checkACLMembership(auth authorization, acl []string) error
 	return errgo.Newf("access denied for user %q", auth.Username)
 }
 
-func (h *ReqHandler) newMacaroon() (*macaroon.Macaroon, error) {
+func (h *ReqHandler) newMacaroon(extraCaveats ...checkers.Caveat) (*macaroon.Macaroon, error) {
```

Comment:
Alternatively, leave this as is and use h.Store.Bakery.AddCaveats in the callers when needed?